F/finufft/build_tarballs.jl

```diff
@@ -36,4 +43,4 @@ dependencies = [
 ]
 
 # Build the tarballs, and possibly a `build.jl` as well.
-build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"8")
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"8", julia_compat=julia_compat)
```

Comment:
Note that if you change `julia_compat` you also _must_ change the version of the package itself, unfortunately the registry won't like having different compatibility bounds when we only change the build number

Reply:
Hmm, but that version is supposed to match the version of the binary library. Or do you suggest that the version of finufft_jll diverge from the version of libfinufft?

Reply:
I know, but some times we do have to diverge, unfortunately.  Or maybe you can try to nicely ping developers to tag a new release, there have been [some changes since last release](https://github.com/flatironinstitute/finufft/compare/v2.0.3...master) already :slightly_smiling_face:

Reply:
I have just pushed 2.0.4. Do I need to tag a release?  I'm waiting on fixing a python wheels CI failure that shouldn't affect anything you're doing.

Reply:
Lovely, thanks a lot! 😃